structure/redblacktree_test/rbtree_test.go

```diff
@@ -0,0 +1,108 @@
+package redblacktree_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/TheAlgorithms/Go/structure/redblacktree"
+)
+
+type RBTree = redblacktree.RBTree
+type RBNode = redblacktree.RBNode
+
+func compare(tree *RBTree, preVal []int, preRed []bool, inVal []int, inRed []bool) bool {
+	// fmt.Println("In compare")
+	var ret = false
+	var pre = redblacktree.PreOrder(tree.Root)
+	var in = redblacktree.InOrder(tree.Root)
+	if len(pre) != len(preVal) || len(in) != len(inVal) {
+		return ret
+	}
+	for i := 0; i < len(pre); i++ {
+		if !pre[i].EquivalentNode(redblacktree.CreateNode(preVal[i], preRed[i])) {
+			fmt.Printf("Error at pre %d", i)
+			return ret
+		}
+	}
+	for i := 0; i < len(in); i++ {
+		if !in[i].EquivalentNode(redblacktree.CreateNode(inVal[i], inRed[i])) {
+			return ret
+		}
+	}
+	return true
+}
+func TestInsert(t *testing.T) {
+	var tree = &RBTree{}
+	var vals = []int{8, 18, 5, 15, 17, 25, 40, 80}
+	var expPreVal = []int{17, 8, 5, 15, 25, 18, 40, 80}
+	var expPreRed = []bool{false, true, false, false, true, false, false, true}
+	var expInVal = []int{5, 8, 15, 17, 18, 25, 40, 80}
+	var expInRed = []bool{false, true, false, false, false, true, false, true}
+	t.Run("Test 1: Insert values", func(t *testing.T) {
+		for _, val := range vals {
+			tree.Insert(val)
+		}
+		if !compare(tree, expPreVal, expPreRed, expInVal, expInRed) {
+			t.Error("Insert did not work")
+		}
+	})
+}
+func TestSearch(t *testing.T) {
```

Comment:
That's the second test, `TestSearch`. It's above the one you've linked here

Reply:
Seems like I went blind here 😓